do/droplets.go

```diff
@@ -154,15 +173,11 @@ func dropletByID(ctx context.Context, client *godo.Client, id string) (*godo.Dro
 		return nil, fmt.Errorf("error converting droplet id to string: %v", err)
 	}
 
-	droplet, resp, err := client.Droplets.Get(ctx, intId)
+	droplet, _, err := client.Droplets.Get(ctx, intId)
 	if err != nil {
 		return nil, err
 	}
 
-	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("DO API returned non-200 status code: %d", resp.StatusCode)
-	}
-
```

Comment:
Removing this code path may cause unexpected side-effects on other code that was expecting errors when the droplet wasn't found. Have you examined those closely to make sure other callers won't panic when `droplet` is nil?

Reply:
yeah, godo returns nil for anything within 200 - 299 anyways and non-nil err otherwise, so it _should_ be the same behaviour. 

https://github.com/digitalocean/godo/blob/master/godo.go#L350-L352

Reply:
if anything, this might be a bug fix cause response may not be `http.StatusOK` but still 2XX